pkg/operator/ceph/cluster/mgr/mgr_test.go

```diff
@@ -108,8 +108,9 @@ func validateStart(t *testing.T, c *Cluster) {
 		}
 		logger.Infof("Looking for cephmgr replica %d", i)
 		daemonName := mgrNames[i]
-		_, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Get(fmt.Sprintf("rook-ceph-mgr-%s", daemonName), metav1.GetOptions{})
+		d, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Get(fmt.Sprintf("rook-ceph-mgr-%s", daemonName), metav1.GetOptions{})
 		assert.Nil(t, err)
+		assert.Equal(t, fmt.Sprint(map[string]string{"my": "annotation"}), fmt.Sprint(d.Spec.Template.Annotations))
```

Comment:
Had you tried to compare the map and annotations directly with the `assert.Equal()` func?

Maybe you can use one of the other `assert.*` functions, ther should be something like `assert.Contains` that may fit.
Or doing two checks, one for does the `my` key exist and second if the value is correct.

Reply:
`assert.ElementsMatch` is the one that seems most appropriate to me. We also want to make sure no extra annotations are added, yes?

Reply:
I believe `assert.ElementsMatch` is for slices
What do you think about doing `assert.True(t, cmp.Equal(map1, map2))` or `assert.True(t, reflect.DeepEqual(map1, map2))` ?

Reply:
as for the contains, I can do that as well, but will need to add the assertion to check lengths of both maps

Reply:
Are you certain that `assert.Equal(t, map1, map2)` doesn't do what we want?

Reply:
To compare the contents of the map it will likely be simplest to do something like this:
```
assert.Equal(t, 1, len(d.Spec.Template.Annotations)
assert.Equal(t, "annotation", d.Spec.Template.Annotations["my"])
```

Reply:
I just ran a test with assert.Equal and it looks to have passed

I think the issue in the past was that assert.Equal also did the Sprintf check before Go1.12 sorted the keys but now looking at the code it uses reflect.DeepEqual under the covers

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L334-L343
https://github.com/stretchr/testify/blob/master/assert/assertions.go#L58-L65